### PR TITLE
Add a dictionary extension to work with KeyValuePair on add

### PR DESCRIPTION
**Pull request by lepsch**

### CODE REVIEW

File: src/Nuhaven.Collections.Generic.Extensions/DictionaryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace Nuhaven.Collections.Generic
+{
+    public static class DictionaryExtensions
+    {
+        /// <summary>
+        /// Adds a key-value-pair instance to a dictionary
+        /// </summary>
+        /// <param name="dict">The IDictionary instance</param>
+        /// <param name="kvp">A key-value-pair to add</param>
+        public static void Add<K, V>(this IDictionary<K, V> dict, KeyValuePair<K, V> kvp) =>
```
Comment (victorbotamedi) [inline src/Nuhaven.Collections.Generic.Extensions/DictionaryExtensions.cs]: List the possible exceptions on the docstring:

ArgumentNullException:	key is null.
ArgumentException: An element with the same key already exists in the IDictionary object.
NotSupportedException: The IDictionary is read-only or the IDictionary has a fixed size.

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionTests.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            const int expectedKey = 1;
+            const string expectedValue = "test";
+
+            var dict = new Dictionary<int, string>();
+
+            Assert.False(dict.ContainsKey(expectedKey));
+            Assert.False(dict.ContainsValue(expectedValue));
+
+            var kvp = new KeyValuePair<int, string>(expectedKey, expectedValue);
+            dict.Add(kvp);
```
Comment (victorbotamedi) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionTests.cs]: Add a test for each possible exception

File: src/Nuhaven.Collections.Generic.Extensions/DictionaryExtensions.cs (hunk 1 above)
Comment (lepsch) [inline src/Nuhaven.Collections.Generic.Extensions/DictionaryExtensions.cs]: :+1:

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionTests.cs (hunk 2 above)
Comment (lepsch) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionTests.cs]: :+1:

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using Xunit;
+
+namespace Nuhaven.Collections.Generic.Extensions.Tests
+{
+    public class DictionaryExtensionsTests
+    {
+        [Fact]
+        public void TestDictionaryAnAddKeyValuePair()
```
Comment (victorbotamedi) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs]: Is this test name correct?

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+            var kvp = new KeyValuePair<int, string>(expectedKey, expectedValue);
+            dict.Add(kvp);
+
+            Assert.True(dict.ContainsKey(expectedKey));
+            Assert.True(dict.ContainsValue(expectedValue));
+        }
+
+        [Fact]
+        public void TestAddingAKeyValuePairWithNullKeyShouldThrow()
```
Comment (victorbotamedi) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs]: What about `TestThrowsIfKeyIsNull`? You are already in a test class about adding a `KeyValuePair` into a `IDictionary`, i think tests names does not need to repeat that over and over

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs (hunk 3 above)
Comment (lepsch) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs]: Didn't know what I was thinking. :stuck_out_tongue:

File: test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs (hunk 4 above)
Comment (lepsch) [inline test/Nuhaven.Collections.Generic.Extensions.Tests/DictionaryExtensionsTests.cs]: About repeating `AddKeyValuePair` over and over it's to discriminate from other tests (future Dictionary extensions)

**Pull request merged**